src/response.rs

```diff
@@ -466,7 +460,9 @@ impl Response {
 
     #[cfg(test)]
     pub fn set_previous(&mut self, previous: Arc<Response>) {
-        self.previous = Some(previous);
+        let mut history = previous.previous.clone();
```

Comment:
> > Right now, each response gets a cloned Vec.

> The vec is not cloned. This is the reason I moved previous out of do_from_request, so I would be able to access it directly even in case of an IO error.

Isn't this (line 463) cloning the vec?

Also, I noticed this is still taking an `Arc<Response>`, but I think it can just be a `&Response` now, right?

Reply:
This is cloning the vec, but it's for tests only, because it takes a `Vec`. The main logic in `connect_inner` doesn't do any clones.

> Also, I noticed this is still taking an Arc<Response>, but I think it can just be a &Response now, right?

I think so, yes. I'll try changing that.